pkg/storage/engine.go

```diff
@@ -387,6 +387,13 @@ type ExportOptions struct {
 	// would stop immediately when targetSize is reached and return the next versions
 	// timestamp in resumeTs so that subsequent operation can pass it to firstKeyTs.
 	StopMidKey bool
+	// ResourceLimiter limits how long iterator could run until it exhausts allocated
+	// resources. Iterator queries limiter in its internal advance loop to break out
+	// of it once resources are exhausted.
+	// This option could only be used together with StopMidKey as it could break
+	// iteration over data on any version and even on a key and timestamp that would
+	// not be included in the resulting export.
```

Comment:
Is this comment up to date? I would expect we don't need to limit this to StopMidKey now that the decision on when to stop is not in the MVCCIterator code.